enhancements/fbc-run-bundle.md

```diff
@@ -0,0 +1,268 @@
+---
+title: fbc-run-bundle
+authors:
+  - "@rashmigottipati"
+reviewers:
+  - "@joelanford"
+  - "@jmrodri"
+  - "@VenkatRamaraju"
+approvers:
+  - "@joelanford"
+  - "@jmrodri"
+  - "@VenkatRamaraju"
+creation-date: 2022-04-20
+last-updated: 2022-04-20
+status: implementable
+see-also:
+replaces:
+superseded-by:
+
+---
+
+# Add FBC support to run bundle
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [X] Test plan is defined
+- [X] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions
+
+- Do we need to build a File-Based Catalog to run this command?
+
+  - We can use `quay.io/operatorhubio/catalog:latest` for testing and generate a File-Based Catalog at runtime when the run bundle command is invoked.
+
+- Do we need to make our own image to use as a default one for run bundle and run bundle-upgrade?
+
+  - Currently, we are using `quay.io/operator-framework/opm:latest` as the default index image. We might have to build an image similar to `quay.io/operatorframework/catalog:latest` which uses opm as the base image and also has a built catalog in it.
+
+- While initializing the FBC, should the channel received from the bundle annotations be used to initialize the default channel name in the `opm init` command?
+
+  - We should be able to handle bundles that don't have channel metadata on the annotations.
+
+- If we don’t get channel information from the bundle, then can we add it to a default channel or to the channel HEAD?
+
+- Can we assume the Channel’s “entries” field be initialized to an array with only one element, i.e., {name: “api-operator.v.0.1”} with no “replaces” field?
+
+  - "replaces" field is required for `run bundle-upgrade` but for run bundle, I think it's ok to assume for the channel entries to have just one entry.
+
+## Summary
+
+The goal of this enhancement is to discuss the impact of the new file based catalog (FBC) in Operator Lifecyle Manager and how it affects the Operator SDK, specifically the existing run-bundle command. This EP discusses the implementation/workflow changes of the `run bundle` command to support the new File-Based Catalog (FBC) index format along with continuing to support SQLite based index images.
```

Comment:
Should not the goal of this EP propose a solution for SDK run bundle adds support to work with FBC format?

Reply:
Yup, that's exactly the goal. To propose a solution to add FBC support in `run bundle` and also continuing to support the SQLite images. Do you suggest rephrasing it @camilamacedo86?

Reply:
wdyt about we be more directly in the EP and then make it less verbose.